lnrpc/rpc.proto

```diff
@@ -430,10 +431,25 @@ service Lightning {
     request to a remote peer. Users are able to specify a target number of
     blocks that the funding transaction should be confirmed in, or a manual fee
     rate to us for the funding transaction. If neither are specified, then a
-    lax block confirmation target is used.
+    lax block confirmation target is used. Each OpenStatusUpdate will return
+    the pending channel ID of the in-progress channel. Depending on the
+    arguments specified in the OpenChannelRequest, this pending channel ID can
+    then be used to manually progress the channel funding flow.
     */
     rpc OpenChannel (OpenChannelRequest) returns (stream OpenStatusUpdate);
 
+    /**
+    FundingStateStep is an advanced funding related call that allows the caller
+    to either execute some preparatory steps for a funding workflow, or
+    manually progress a funding workflow. The primary way a funding flow is
+    identified is via its pending channel ID. As an example, this method can be
+    used to specify that we're expecting a funding flow for a particular
+    pending channel ID, for which we need to use specific parameters.
+    Alternatively, this can be used to interactively drive PSBT signing for
+    funding for partially complete funding transactions.
+    */
+    rpc FundingStateStep(FundingTransitionMsg) returns (FundingStateStepResp);
```

Comment:
Commit doesn't compile without implementation:
```
./rpcserver.go:495:5: cannot use (*rpcServer)(nil) (type *rpcServer) as type lnrpc.LightningServer in assignment:
        *rpcServer does not implement lnrpc.LightningServer (missing FundingStateStep method)
```

Reply:
I don't think we've ever made that a requirement for proto changes, as IMO it's nice to have commits that only touch `lnrpc`, then do the actual implementation as a follow up.

Reply:
It's preferable to just add a stub `FundingStateStep` method IMO, since it makes bisecting much more powerful if each commit compiles.